prow/cmd/checkconfig/main.go

```diff
@@ -147,36 +153,52 @@ func main() {
 }
 
 func validateTideRequirements(configAgent config.Agent, pluginAgent plugins.PluginAgent) error {
-	lgtmPluginConfig := newOrgRepoConfig(pluginAgent.Config().EnabledReposForPlugin(lgtm.PluginName))
-	approvePluginConfig := newOrgRepoConfig(pluginAgent.Config().EnabledReposForPlugin(approve.PluginName))
-
-	lgtmTideConfig, approveTideConfig, overallTideConfig := newOrgRepoConfig([]string{}, []string{}), newOrgRepoConfig([]string{}, []string{}), newOrgRepoConfig([]string{}, []string{})
-	for _, query := range configAgent.Config().Tide.Queries {
-		requiresLgtm := sets.NewString(query.Labels...).Has(lgtm.LgtmLabel)
-		requiresApproval := sets.NewString(query.Labels...).Has(approve.ApprovedLabel)
-		for _, org := range query.Orgs {
-			if requiresLgtm {
-				lgtmTideConfig.orgs.Insert(org)
-			}
-			if requiresApproval {
-				approveTideConfig.orgs.Insert(org)
-			}
-			overallTideConfig.orgs.Insert(org)
-		}
-		for _, repo := range query.Repos {
-			if requiresLgtm {
-				lgtmTideConfig.repos.Insert(repo)
-			}
-			if requiresApproval {
-				approveTideConfig.repos.Insert(repo)
+	requires := func(label string, query config.TideQuery) bool {
+		return sets.NewString(query.Labels...).Has(label)
+	}
+	forbids := func(label string, query config.TideQuery) bool {
+		return sets.NewString(query.MissingLabels...).Has(label)
+	}
+
+	configs := []struct {
+		plugin, label string
+		matches       func(label string, query config.TideQuery) bool
+		config        orgRepoConfig
+	}{
+		{plugin: lgtm.PluginName, label: lgtm.LgtmLabel, matches: requires},
+		{plugin: approve.PluginName, label: approve.ApprovedLabel, matches: requires},
+		{plugin: hold.PluginName, label: hold.Label, matches: forbids},
+		{plugin: wip.PluginName, label: wip.Label, matches: forbids},
+		{plugin: verifyowners.PluginName, label: verifyowners.InvalidOwnersLabel, matches: forbids},
+		{plugin: releasenote.PluginName, label: releasenote.ReleaseNoteLabelNeeded, matches: forbids},
+		{plugin: cherrypickunapproved.PluginName, label: cherrypickunapproved.CpUnapprovedLabel, matches: forbids},
+		{plugin: blockade.PluginName, label: blockade.BlockedPathsLabel, matches: forbids},
+	}
+
+	overallTideConfig := newOrgRepoConfig([]string{}, []string{})
+	for _, pluginConfig := range configs {
+		pluginConfig.config = newOrgRepoConfig([]string{}, []string{})
+		for _, query := range configAgent.Config().Tide.Queries {
+			if pluginConfig.matches(pluginConfig.label, query) {
+				for _, org := range query.Orgs {
+					pluginConfig.config.orgs.Insert(org)
+					overallTideConfig.orgs.Insert(org)
```

Comment:
This will only include orgs for which some pluginConfig matches instead of including all orgs.
The same applies to `overallTideConfig.Repos.Insert(repo)` below.

Reply:
Nice catch!